test/stdlib/StringAPI.swift

```diff
@@ -471,5 +490,15 @@ StringTests.test("Regression/corelibs-foundation") {
   expectEqual(substring(of: s5, with: NSFakeRange(1,6)), "\ncats�")
 }
 
+StringTests.test("Regression/radar-87371813") {
```

Comment:
Using @lorentey's test case here to make sure we hit the specific problem in question rather than just relying on the newly generalized test to catch it

Reply:
This is good, but we should also have a test that uses `checkHashable` for this stuff

Reply:
We do! The other test code I added has checkHashable on substrings of each string we checkHashable on

Reply:
💯 Oh indeed, it's right there. Sorry I missed it! 🤦‍♂️